src/tap.h

```diff
@@ -33,6 +33,8 @@
 #include <map>
 #include <fstream>
 
+#include <boost/algorithm/string/replace.hpp>
```

Comment:
Looks like this is being included for boost::replace_all_copy for a simple string replacement. Do you think it would be possible to remove this extra dependency and maybe write a simple method for that? :-)